application-collabora-default/src/main/java/com/xwiki/collabora/internal/rest/DefaultWopi.java

```diff
@@ -151,7 +151,10 @@ public Response postContents(String fileId, String token, byte[] body) throws XW
     {
         if (fileTokenManager.isInvalid(token) || !fileTokenManager.hasAccess(token)) {
             logger.warn("Failed to update file [{}] due to invalid token or restricted rights.", fileId);
-            throw new WebApplicationException(Response.Status.UNAUTHORIZED);
+            // As the cause of a failure in updating the content may be an expired token, we return 200 status code.
+            // In a following request, an extension to the expiration date of the token will be attempted. In case of
+            // failure, the corresponding UNAUTHORIZED message will be returned.
```

Comment:
```
As the cause of a failure in updating the content may be an expired token, we return 200 status code for now since the UNAUTHORIZED message should be returned after trying first to extend the token.
For this, subsequently to this request, an attempt to extend the token validity is done and only if this is not possible (e.g. due to insufficient rights) the correct UNAUTHORIZED message is returned.
This is needed since we couldn't find a way to renew the token before the save request done by Collabora.
```
I've done some reformulation, it looks clearer to me. WDYT?